### PR TITLE
added toleration support for deamonset with same convention

### PR DESCRIPTION
**Pull request by Sajadorouji**
In environments with multiple node pools that have taint and toleration, deamonsets cannot run up on those nodes.

### CODE REVIEW

**Comment by patlachance** [comment]
Thanks, anything else needed to merge this PR?

**Review by ibuziuk — COMMENTED** [review]
Thank you very much for the PR. Could you please also provide the PR for documentation https://www.eclipse.org/che/docs/che-7/administration-guide/caching-images-for-faster-workspace-start/
The docs repo - https://github.com/eclipse/che-docs

**Comment by ibuziuk** [comment]
also, please submit the Eclipse Contributor Agreement - https://www.eclipse.org/legal/ECA.php
This is a pre-requirement for getting the PRs merged

File: cfg/config_test.go [hunk 1]
```diff
@@ -44,6 +46,7 @@ func TestEnvVars(t *testing.T) {
 				"NAMESPACE":           "my-namespace",
 				"NODE_SELECTOR":       "{\"type\": \"compute\"}",
 				"CACHING_CPU_REQUEST": ".055",
+				"NODE_TOLERATION":"[{\"effect\": \"NoSchedule\",\"key\": \"app\",\"operator\": \"Equal\",\"value\": \"prod\"}]",
```
Comment (amisevsk) [inline cfg/config_test.go:49]: Consider using a raw string here:
```suggestion
				"NODE_TOLERATION":`[{"effect": "NoSchedule","key": "app","operator": "Equal","value": "prod"}]`,
```

**Review by amisevsk — APPROVED** [review]
I'm not set up to test changes right now, but generally looks good to me, one comment (and thanks for catching a dirty `go.mod` :smile:!)

**Comment by ibuziuk** [comment]
@Sajadorouji we switched the repository to the 'main' branch, and this PR was closed automatically. We would really appreciate if you could send a new PR to the 'main' branch and sign Eclipse Contributor Agreement so that we would be able to merge it - https://www.eclipse.org/legal/ECA.php

**Comment by charlspjohn** [comment]
@Sajadorouji , It would be great if you could create a new PR against `main` branch.
Thanks in advance !

**Pull request closed without merging**